master/getting-started/kubernetes/upgrade.md

```diff
@@ -167,3 +172,33 @@ To upgrade the policy controller:
 We recommend running the policy controller as a Kubernetes Deployment with type "recreate", in which
 case upgrade can be handled entirely through the
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 (when using the Kubernetes datastore driver) interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).  If your
+Calico deployment uses the etcd datastore driver, there is no change in behavior in v2.3.0+ (this change is planned for the v2.4.0 release).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow these steps prior to upgrading Calico to ensure policy is
+enforced the same throughout the upgrade process.
```

Comment:
"ensure policy is enforced the same" feels grammatically weird to me.  Perhaps "ensure policy enforcement does not change"?